core/src/test/java/com/linecorp/armeria/server/throttling/BlockingTaskLimitingThrottlingStrategyTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.throttling;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.ContextAwareScheduledExecutorService;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.util.SettableIntSupplier;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+class BlockingTaskLimitingThrottlingStrategyTest {
+    @Test
+    public void blockingTaskLimiting() {
+        final HttpRequest request = HttpRequest.of(HttpMethod.GET, "/");
+        final ServiceRequestContext ctx = mock(ServiceRequestContext.class);
```

Comment:
Could you use a real object using `ServiceRequestContext.builder(...)` rather than mocking the responses?
The mocking objects will pass assertions even if the actual code goes wrong.

Reply:
Could you tell me in more datil about `if the actual code goes wrong`?

Reply:
The following code may raise a `ClassCastException` with the default `BlockingTaskExecutor`.
```java
final ThreadPoolExecutor executor = (ThreadPoolExecutor) ctx.blockingTaskExecutor().withoutContext(); 
```
 However, this test seems not to detect the error.

Reply:
Is there a way to create `Server` (may using `ServerExtension`?) and queuing requests in server for testing?

Reply:
> Is it good with delaying the requests for a while?.?

I'm not sure which scenario you want to make, but I think it's the way to go.
We can create an `HttpService` that delays the response and check the limit.

Reply:
Thank you 😄